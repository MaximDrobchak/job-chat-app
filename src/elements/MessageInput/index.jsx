import React from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';
import Button from '../Button';
import attachmentImage from '../../assets/attachment.svg';
const styles = {
	root: {
		position: 'relative',
		'& input': {
			width: '100%',
			height: 46,
			borderRadius: 4,
			backgroundColor: '#fff',
			border: 'solid 2px #ecedf0',
			fontFamily: 'Roboto-Regular',
			fontSize: 18,
		},
	},
	buttonMessage: {
		position: 'absolute',
		right: '1%',
		top: '25%',
	},
};
const MessageInput = ({ classes }) => (
	<div className={classes.root}>
		<input
			// className={classes.inputMessage}
			type="text"
			placeholder="Message to Terry Crews"
		/>
		<Button className={classes.buttonMessage}>
			<img src={attachmentImage} alt="attachment" />
		</Button>
	</div>
);

MessageInput.propTypes = {
	classes: PropTypes.object.isRequired,
};
export default injectSheet(styles)(MessageInput);
