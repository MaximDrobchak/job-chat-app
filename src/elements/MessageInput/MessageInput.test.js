import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from './index.jsx';

describe('MessageInput', () => {
	it('отрисовывает без ошибки', () => {
		const div = document.createElement('div');
		ReactDOM.render(<MessageInput />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('есть корректный снимок', () => {
		const component = renderer.create(<MessageInput />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
