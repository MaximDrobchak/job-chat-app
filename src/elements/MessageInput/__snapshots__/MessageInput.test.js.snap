// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput есть корректный снимок 1`] = `
<div
  className="MessageInput-root-0-1-1"
>
  <input
    placeholder="Message to Terry Crews"
    type="text"
  />
  <span
    className="MessageInput-buttonMessage-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="attachment"
      src="attachment.svg"
    />
  </span>
</div>
`;
